packages/cdktf/lib/annotations.ts

```diff
@@ -0,0 +1,79 @@
+// Originally from aws-cdk v2 because with constructs v10 addMessage was removed
+// in favor of Annotations in the AWS CDK
+// https://github.com/aws/aws-cdk/blob/dcae3eead0dbf9acb1ed80ba95bb104c64cb1bd7/packages/@aws-cdk/core/lib/annotations.ts
+import { IConstruct } from "constructs";
+
+export enum AnnotationMetadataEntryType {
```

Comment:
I wonder if it would make sense to use the same values as AWS CDK for this and other pieces in an attempt for future interop support.

Reply:
I thought about that too. In general these are the same levels (info, warn, error) and I think a mapping would be easy enough for a future interop support. I would not want to pull in the AWS CDK as a dependency just for this now.
While using the `Aspect` from the AWS CDK should work, the `Annotations` class contains code that is dependent on other AWS CDK constants which would open a can of worms I think.
And having a hard dependency on the AWS CDK for CDKTF internals might be bad in the future if we want to support a range of AWS CDK constructs which themselves could need different AWS CDK core versions (just a suspicion at this point). Building an adapter would probably yield better results then. Similar to this poc: https://github.com/skorfmann/cfn2tf/blob/interop-poc/aws-adapter.ts